### PR TITLE
validate input data

### PR DESCRIPTION
**Pull request by hcvcastro**
- browser: add method 'setContainer' to JSDialog
- browser: use 'checkValidity' for spin field element
- browser: use 'reportValidity' to validate number

* Resolves: # 
* Target version: master 

### TODO

- [ ] ...

### Checklist

- [ ] Code is properly formatted
- [ ] All commits have Change-Id
- [ ] I have run tests with `make check`
- [ ] I have issued `make run` and manually verified that everything looks okay
- [ ] Documentation (manuals or wiki) has been updated or is not required

### CODE REVIEW

**Comment by hcvcastro** [comment]
prevent send invalid input like 5.6- or out out range min, max values.

**Comment by hcvcastro** [comment]
Done

File: browser/src/control/Control.JSDialogBuilder.js [hunk 1]
```diff
@@ -323,10 +342,12 @@ L.Control.JSDialogBuilder = L.Control.extend({
 
 	// by default send new state to the core
 	_defaultCallbackHandler: function(objectType, eventType, object, data, builder) {
-
 		if (builder.map.uiManager.isUIBlocked())
 			return;
 
+		if (objectType === 'responsebutton' && data === '1' && !builder.reportValidity())
```
Comment (eszkadev) [inline browser/src/control/Control.JSDialogBuilder.js:348]: now it doesn't work,    type of data is a Number I think, not string

**Comment by hcvcastro** [comment]
Done

**Review by eszkadev — APPROVED** [review]

**Comment by hcvcastro** [comment]
The unit test fail impress/apply_paragraph_props_text_spec.js is unrelated to this pull request

**Pull request merged**